alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/model/action/Throttle.kt

```diff
@@ -0,0 +1,84 @@
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.model.action
+
+import org.apache.commons.codec.binary.StringUtils
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.ToXContentObject
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+import java.lang.IllegalStateException
+import java.time.temporal.ChronoUnit
+import java.util.Locale
+
+data class Throttle(
+    val value: Int?,
+    val unit: ChronoUnit = ChronoUnit.MINUTES
+) : ToXContentObject {
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        return builder.startObject()
+                .field(VALUE_FIELD, value)
+                .field(UNIT_FIELD, unit.name)
+                .endObject()
+    }
+
+    companion object {
+        const val VALUE_FIELD = "value"
+        const val UNIT_FIELD = "unit"
+
+        @JvmStatic
+        @Throws(IOException::class)
+        fun parse(xcp: XContentParser): Throttle {
+            var value: Int? = null
+            var unit: ChronoUnit? = null
+
+            XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.currentToken(), xcp::getTokenLocation)
+            while (xcp.nextToken() != XContentParser.Token.END_OBJECT) {
+                val fieldName = xcp.currentName()
+                xcp.nextToken()
+                when (fieldName) {
+                    UNIT_FIELD -> {
+                        val unitString = xcp.text().toUpperCase(Locale.ROOT)
+                        require(StringUtils.equals(unitString, ChronoUnit.MINUTES.name), { "Only support MINUTES throttle unit" })
+                        unit = ChronoUnit.valueOf(unitString)
+                    }
+                    VALUE_FIELD -> {
+                        val currentToken = xcp.currentToken()
+                        when {
+                            currentToken == XContentParser.Token.VALUE_NULL -> value = null
+                            currentToken.isValue -> {
+                                value = xcp.intValue()
+                                require(value > 0, { "Can only set positive throttle period" })
+                            }
+                            else -> {
+                                XContentParserUtils.throwUnknownToken(currentToken, xcp.tokenLocation)
+                            }
+                        }
+                    }
+
+                    else -> {
+                        throw IllegalStateException("Unexpected field: $fieldName, while parsing action")
+                    }
+                }
+            }
+
+            return if (unit == null) Throttle(value = value) else Throttle(value = value, unit = unit)
```

Comment:
This allows someone to do { throttle: { value: 1 } }, why not require unit? I see it'll default to MINUTEs when not specified, but why even allow it?

Reply:
Here the unit is kept for the flexibility to support more options like HOURS

Reply:
I'm confused why requiring unit explicitly would stop us from allowing more options like HOURS in the future?